src/main/java/com/movie/store/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -2,25 +2,33 @@
 
 import com.movie.store.dao.CinemaHallDao;
 import com.movie.store.exceptions.DataProcessingException;
-import com.movie.store.lib.Dao;
 import com.movie.store.model.CinemaHall;
-import com.movie.store.util.HibernateUtil;
 import java.util.List;
 import javax.persistence.criteria.CriteriaQuery;
+import lombok.extern.log4j.Log4j;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Log4j
+@Repository
+public class CinemaHallDaoImpl extends AbstractSessionFactoryCreator implements CinemaHallDao {
```

Comment:
it's not logical that Dao extends SessionFactoryCreator + you still call constructor in each class, so class AbstractSessionFactoryCreator is redundant, let's remove it